python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -410,6 +410,18 @@ def submit(
     def __repr__(self):
         return f"MapWorker({self.src_fn_name})"
 
+    def on_exit(self):
+        """Called when the actor is about to exist.
+        This enables performing cleanup operations via `UDF.__del__`.
+
+        Note, this only ensures cleanup is performed when the job exists gracefully.
+        If the driver or the actor is forcefully killed, `__del__` will not be called.
+        """
+        # `_map_actor_context` is a global variable that references the UDF object.
+        # Delete it to trigger `UDF.__del__`.
+        del ray.data._map_actor_context
+        ray.data._map_actor_context = None
+
```

Comment:
Is there a reason we need to introduce an explicit method rather than overriding `_MapWorker.__del__`? I imagine if we override `__del__`, we'll implicitly do the cleanup when we run `del self._running_actors[actor]`.

Reply:
I also tried that. But it looks like that Ray Core doesn't call `Actor.__del__` when killing an actor. cc @jjyao

Reply:
ok, I found the ray core issue https://github.com/ray-project/ray/issues/50048 cc @jjyao 
I can fix it if the solution looks good to you.